### PR TITLE
Little completion/doc improvements

### LINKED ISSUE #21: completion after a package is unintuitive
For example (`|` is the cursor.)

``` go
package main

import (
    "go/parser"
)

func main() {
    pars|
}
```

Currently if you complete at that point and move down to `package parser` it will look like

``` go
package main

import (
    "go/parser"
)

func main() {
    parser.|
}
```

But now you have to first press `<c-y>` to accept the completion before the completion can continue on the functions of the `parser` package.

Instead after completing `package parser` it should not insert a period. Instead:

``` go
package main

import (
    "go/parser"
)

func main() {
    parser|
}
```

Then type the period and all is good. Its much more intuitive then pressing `<c-y>`.

### PR DESCRIPTION
**Pull request by nhooyr**
This fixes #21, #20 and #17

Option for period after package completion. Option for aligned classes.
Type info for vars/types is now shown. Gocode's offset is 1 behind
neovim's. Most of the documentation was also updated.

This is a continuation of #22. Sorry I accidentally did something with the old branch that led to the pull request closing.

### CODE REVIEW

**Comment by zchee** [comment]
@nhooyr Good morning :) and Thanks!

I tested.
type info, on/off dot completion and alignment, sort_class, they are all successfully.
and, update `README.md` is amazing! screenshot is easy to understand.

but, there is a point to be worried about.

---

When you write in its place, will get the user settings each time generate a completion word.
- https://github.com/zchee/deoplete-go/pull/23/files#diff-8a8a0b088ea59e8673b0b7060d0c39c6R73
- https://github.com/zchee/deoplete-go/pull/23/files#diff-8a8a0b088ea59e8673b0b7060d0c39c6R74 

I think it's better to write to the `__init__` section.
simple verify code.

``` python
    def __init__(self, vim):
        .
        .
        self.test = 0
        self.vim.vars['deoplete#sources#go#test'] = self.test
        self.test += 1
```

It will be `deoplete#sources#go#test` is always `0`.
but

``` python
    def __init__(self, vim):
        .
        .
        self.test = 0

        .
        .
        try:
            out = []
            self.vim.vars['deoplete#sources#go#test'] = self.test
            self.test += 1
            align_class = self.vim.vars['deoplete#sources#go#align_class']
            package_dot = self.vim.vars['deoplete#sources#go#package_dot']
```

It will increase.

Why remove `Exception`?
https://github.com/zchee/deoplete-go/pull/23/files#diff-8a8a0b088ea59e8673b0b7060d0c39c6L32
It will be Flake8 is suggestions `no 'except:' at least use 'except: Exception:'`

and, sorry nitpick.
I do not like to put things that are not related to the code in the root, could you create `images` dir and move to `align_class.png` into `images`? :bow:

**Comment by nhooyr** [comment]
Absolutely. I understand.

**Comment by nhooyr** [comment]
@zchee should be good now.

**Comment by zchee** [comment]
@nhooyr Thanks for the fix!
I'm continue to test.
If there are no problems, and then merge.

**Comment by nhooyr** [comment]
Thats good. One question.
What does this part of the readme mean.

`If you want to the same motion as the omnifunc, set`

``` vim
call deoplete#custom#set('go', 'min_pattern_length', 1000)
```

**Comment by zchee** [comment]
@nhooyr The details is here. https://github.com/zchee/deoplete-go/pull/8#issuecomment-173676614
but I do not use this setting, so I do not know whether still valid.

/cc @fatih
If you have time, Could you test it?

File: README.md [hunk 1]
```diff
@@ -24,19 +24,32 @@ Plug 'zchee/deoplete-go', { 'do': 'make'}
...
-Available values is `package`, `func`, `type`, `var`, `const`.  
-`deoplete#sources#go#sort_class` **value must be include all class.**
+First is unaligned, second is aligned.
+
+<img src="https://raw.githubusercontent.com/zchee/deoplete-go/master/images/align_class.png" width="600">
+
+If you would like them aligned just set:
+
+```vim
+let g:deoplete#sources#go#align_class = 1
```
Comment (fatih) [inline README.md:37]: I think this is not needed. Why? The more settings you have the more complex will your app. Always try to keep it simple. I don't think an alignment is needed but it's your call :)

Comment (fatih) [inline README.md:37]: Alright seems like the alignment is only for the keywords. So this gets a +1 from me then.

**Comment by nhooyr** [comment]
@zchee I don't think that option is necessary anymore. I just removed it.

File: README.md (hunk 1 above)
Comment (zchee) [inline README.md:37]: @fatih 

> Alright seems like the alignment is only for the keywords.

Are keywords means `class`?
alignment for `class` only. available `package`, `func`, `type`, `var`, `const`.
but `package` and `[func`, `type`, `var`, `const]` almost does not display at the same time.
So alignment of `6`.

Comment (nhooyr) [inline README.md:37]: ah i'll remove the package part, its not aligned. only those 4 are.

**Comment by nhooyr** [comment]
Whoops I just found a pretty big bug. I have to rename ftplugin to plugin. The go source is initialized even if its not a go file. otherwise we get nasty errors because the variables for the settings aren't defined.

File: README.md (hunk 1 above)
Comment (fatih) [inline README.md:37]: For example this is my sort_class:

```
  let g:deoplete#sources#go#sort_class = ['func', 'type', 'var', 'const']
```

So is my alignment 6 (`len(const) + 1`) or 8 (`len(package) + 1`) ?

Comment (nhooyr) [inline README.md:37]: it is 6. It is always 6 no matter what. We don't align with packages because the space is way too big.

Comment (fatih) [inline README.md:37]: Thanks for the clarification, so it makes sense then :+1:

**Comment by zchee** [comment]
@nhooyr OK, I wait for fix.

but, 

> I don't think that option is necessary anymore. I just removed it.

I need a completion of dot, You alignment is not necessary.
As of them, will each have preference settings.
Because even wrote to `README.md` performance does not slow, do you leave for them?

**Comment by nhooyr** [comment]
@zchee i fixed it.

completion of dot? Ohh I meant this part of the readme.

`If you want to the same motion as the omnifunc, set`

``` vim
call deoplete#custom#set('go', 'min_pattern_length', 1000)
```

This part isn't necessary anymore. Not talking about any options, sorry, i just meant that part of the readme.

**Comment by zchee** [comment]
@nhooyr No, no!
I need a completion of dot, your alignment is not necessary. @ svanharmelen(no mention) need a `call deoplete#custom#set('go', 'min_pattern_length', 1000)` beheiver.

...I think so, but certainly the setting does not work.
Yes, remove it.

**Comment by nhooyr** [comment]
@zchee i didn't remove completion of the dot. its still there. i'm confused. do you mean by default you want the dot completed?

i only removed that part from the README.

**Comment by zchee** [comment]
@nhooyr Ah, It is a figuratively. example.
We have a plugins behavior of their own preferences. So if still work this setting, I thought to want to leave.

but, I test `call deoplete#custom#set('go', 'min_pattern_length', 1000)` now.
It's no longer not work in current code.
As a result, I agreed for

> I don't think that option is necessary anymore. I just removed it.

Sorry poor English :(

**Comment by nhooyr** [comment]
Its good. Gotcha.

**Comment by nhooyr** [comment]
Well I think I'm done testing. LGTM

File: README.md (hunk 1 above)
Comment (zchee) [inline README.md:37]: @fatih :)

**Comment by zchee** [comment]
@nhooyr Now the time of breakfast in Japan. :sunny:
Please wait a moment :)

**Comment by nhooyr** [comment]
Oh wait I'm gonna fix a quick problem with screenshot linking.

**Comment by zchee** [comment]
@nhooyr I got it.

**Comment by nhooyr** [comment]
@zchee does the image look too big? https://github.com/nhooyr/deoplete-go/tree/completion-improvements

**Comment by zchee** [comment]
@nhooyr No. size is LGTM.

But I use the display's size of `3840x2160`.
My opinion be useless...

**Comment by nhooyr** [comment]
@fatih what do you think of the image size here https://github.com/nhooyr/deoplete-go/tree/completion-improvements

**Comment by nhooyr** [comment]
Well it doesn't look bad so I think its all good.

**Comment by zchee** [comment]
@nhooyr I also done testing.
LGTM.

**Comment by nhooyr** [comment]
@zchee just go ahead and merge then. It looks good.

**Comment by zchee** [comment]
@nhooyr @fatih Merged.
Thanks for discuss and cooperation!

**Comment by nhooyr** [comment]
@zchee no problemo

**Pull request merged**